src/integrations/ditto/client/APIDitto.ts

```diff
@@ -0,0 +1,79 @@
+import { buildClient, sendGet, sendPut } from '@lokalise/node-core'
+import type { Client } from 'undici'
+
+import type { Dependencies } from '../../../infrastructure/diConfig'
+import { AuthFailedError, UnrecognizedError } from '../../../infrastructure/errors/publicErrors'
+
+import type {
+  UpdateVariantRequestBody,
+  UpdateVariantResult,
+  WorkspaceComponentsByName,
+} from './types'
+
+export class APIDitto {
+  private readonly client: Client
+
+  constructor({ config }: Dependencies) {
+    const host = config.integrations.ditto.baseUrl
+    if (!host) {
+      throw new Error('Ditto API path not provided')
+    }
+    this.client = buildClient(host, {
+      bodyTimeout: 5000,
+      headersTimeout: 5000,
+    })
+  }
+
+  private extractHeaders(apiKey: string) {
+    return {
+      Authorization: `token ${apiKey}`,
+      origin: 'lokalise',
+      'Content-Type': 'application/json',
+    }
+  }
+
+  private extractRetryConfig() {
```

Comment:
doesn't need to be a method, can easily be just a constant

Reply:
changed anyway